docs/advanced/compiler-annotations.md

```diff
@@ -665,3 +509,244 @@ end
+
+## @metaExtension
+
+<DeprecatedInVersion deprecated="0.37.0" removed="TBD" />
+
+**Target elements:** `class`
+
+The Extension decorator marks a class as an extension of an already existing meta class/table.
+This causes the class header to not be translated, preventing instantiation and the override of the existing class.
+
+**Example**
+
+<SideBySide>
+
+```typescript
+class MyBaseClass {
+  myFunction(): void {}
+}
+```
+
+```lua
+MyBaseClass = __TS__Class()
+...
+function MyBaseClass.prototype.myFunction(self) end
+```
+
+</SideBySide>
+
+<SideBySide>
+
+```typescript
+/** @metaExtension */
+class MyMetaExtension extends MyMetaClass {
+  myFunction(): void {}
+}
+```
+
+```lua
+local __meta__MyMetaClass = debug.getregistry().MyMetaClass
+__meta__MyMetaClass.myFunction = function(self)
+end;
+```
+
+</SideBySide>
+
+**Upgrade Instructions**
+
+Use an interface to extend your existing class and assign the functions to the meta table of the existing class.
+
+<SideBySide>
+
+```typescript
+interface MyMetaExtension extends MyMetaClass {
+  myFunction(): void;
+}
+
+const MyMetaExtensionTable: MyMetaExtension = debug.getregistry().MyMetaClass as MyMetaExtension;
+
+MyMetaExtensionTable.myFunction = function () {};
+```
+
+```lua
+MyMetaExtensionTable = debug.getregistry().MyMetaClass
+MyMetaExtensionTable.myFunction = function(self)
+end
+```
+
+</SideBySide>
+
+## @phantom
+
+<DeprecatedInVersion deprecated="0.37.0" removed="TBD" />
+
+**Target elements:** `namespace`
+
+This decorator marks a namespace as a phantom namespace.
+This means all members of the namespace will be translated as if they were not in that namespace. Primarily used to prevent scoping issues.
+
+**Example**
+
+<SideBySide>
+
+```typescript
+namespace myNameSpace {
+  function myFunction(): void {}
+}
+```
+
+```lua
+myNameSpace = {}
+function myNameSpace.myFunction() end
+```
+
+</SideBySide>
+
+<SideBySide>
+
+```typescript
+/** @phantom */
+namespace myNameSpace {
+  function myFunction(): void {}
+}
+```
+
+```lua
+function myFunction() end
+```
+
+</SideBySide>
+
+## @pureAbstract
+
+<DeprecatedInVersion deprecated="0.37.0" removed="TBD" />
+
+**Target elements:** `declare class`
+
+This decorator marks a class declaration as purely abstract.
+The result is that any class extending the purely abstract class will not extend this class in the resulting Lua.
+
+**Example**
+
+<SideBySide>
+
+```typescript
+declare class MyAbstractClass {}
+class MyClass extends MyAbstractClass {}
+```
+
+```lua
+MyClass = __TS__Class()
+MyClass.__base = MyAbstractClass
+MyClass.____super = MyAbstractClass
+setmetatable(MyClass, MyClass.____super)
+setmetatable(MyClass.prototype, MyClass.____super.prototype)
+```
+
+</SideBySide>
+
+<SideBySide>
+
+```typescript
+/** @pureAbstract */
+declare class MyAbstractClass {}
+class MyClass extends MyAbstractClass {}
+```
+
+```lua
+MyClass = __TS__Class()
+```
+
+</SideBySide>
+
+**Upgrade Instructions**
+
+Use interface merging.
```

Comment:
Maybe mention preferred is to declare interfaces over classes, otherwise use the the TS below.